### PR TITLE
Add "How to use ROCm for AI"

### PR DESCRIPTION
**Pull request by peterjunpark**
D3
This PR adds a tutorial-like guide on common usage patterns re: using ROCm with deep learning frameworks for training a model, optimizing development, basic fine-tuning, inference and serving.

### CODE REVIEW

**Review by samjwu — CHANGES_REQUESTED** [review]
https://rocm.docs.amd.com/projects/rocm-docs-core/en/latest/user_guide/spellcheck.html#

this is a user guide for spell check and how to fix flagged errors

File: docs/how-to/rocm-for-ai/train-a-model.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+   :keywords: ROCm, AI, LLM, train, fine-tune, FSDP, DeepSpeed, LLaMA, tutorial
+
+****************
+Training a model
+****************
+
+The following is a brief overview of popular component paths per AI development use-case, such as training, LLMs,
+and inferencing.
+
+Accelerating model training
```
Comment (peterjunpark) [inline docs/how-to/rocm-for-ai/train-a-model.rst:12]: Do we need more content on the basics of training a model with ROCm before jumping straight into optimization with distributed training?

File: docs/how-to/rocm-for-ai/index.rst [hunk 2]
```diff
@@ -0,0 +1,23 @@
+.. meta::
+   :description: How to use ROCm for AI
+   :keywords: ROCm, AI, machine learning, LLM, usage, tutorial
+
+**********************
+How to use ROCm for AI
```
Comment (lpaoletti) [inline docs/how-to/rocm-for-ai/index.rst:6]: This should match the TOC (left-hand panel)

File: docs/how-to/rocm-for-ai/install.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+      Follow the :doc:`post-installation instructions <rocm-install-on-linux:how-to/native-install/post-install>` to
+      configure your system linker, PATH, and verify the installation.
+
+      If you encounter any issues during installation, refer to the
+      :doc:`Installation troubleshooting <rocm-install-on-linux:how-to/native-install/install-faq>` guide.
+
+ROCm supports popular machine learning frameworks including `PyTorch
+<https://pytorch.org/blog/pytorch-for-amd-rocm-platform-now-available-as-python-package>`_, TensorFlow, JAX, MAGMA, and
+DeepSpeed.
```
Comment (lpaoletti) [inline docs/how-to/rocm-for-ai/install.rst:47]: On the next page (Training...), you have a hyperlink on DeepSpeed. Perhaps put it here, too. 
By the way, why is DS the only other framework mentioned on the next page with PyTorch.

File: docs/how-to/rocm-for-ai/hugging-face-models.rst [hunk 4]
```diff
@@ -0,0 +1,205 @@
+.. meta::
+   :description: How to use ROCm for AI
+   :keywords: ROCm, AI, LLM, Hugging Face, Optimum, Flash Attention, GPTQ, ONNX, tutorial
+
+********************************
+Running models from Hugging Face
+********************************
+
+`Hugging Face <https://huggingface.co>`_ hosts the world’s largest AI model repository for developers to obtain
+state-of-the-art transformer models. This section describes how to run the popular community transformer models from
```
Comment (lpaoletti) [inline docs/how-to/rocm-for-ai/hugging-face-models.rst:10]:
```suggestion
transformer models. This section describes how to run the popular community transformer models from
```

File: docs/how-to/rocm-for-ai/hugging-face-models.rst [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+   pip install --upgrade --upgrade-strategy eager optimum[amd]
+
+Or, install from source.
+
+.. code-block:: shell
+
+   git clone https://github.com/huggingface/optimum-amd.git
+   cd optimum-amd
+   pip install -e .
```
Comment (lpaoletti) [inline docs/how-to/rocm-for-ai/hugging-face-models.rst:88]: Should the period be at the end of the line?

File: docs/how-to/rocm-for-ai/index.rst [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+In this guide, you'll learn about:
+
+- :doc:`Installing ROCm and machine learning frameworks <install>`
+
+- :doc:`Training a model <train-a-model>`
+
+- :doc:`Deploying your model <deploy-your-model>`
+
+- :doc:`Running community models from Hugging Face <hugging-face-models>`
```
Comment (lpaoletti) [inline docs/how-to/rocm-for-ai/index.rst:23]:
```suggestion
- :doc:`Running models from Hugging Face <hugging-face-models>`
```

File: docs/how-to/rocm-for-ai/train-a-model.rst [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+As of PyTorch 1.6.0, features in `torch.distributed` are categorized into three main components:
+
+- `Distributed data-parallel training
+  <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`_ (DDP)
+
+- `RPC-Based distributed training <https://pytorch.org/docs/stable/rpc.html>`_ (RPC)
+
+- `Collective communication <https://pytorch.org/docs/stable/distributed.html>`_
+
+In this guide, we focus on the distributed data-parallelism strategy as it’s the most popular. To get started with DDP,
```
Comment (lpaoletti) [inline docs/how-to/rocm-for-ai/train-a-model.rst:33]:
```suggestion
In this guide, the focus is on the distributed data-parallelism strategy as it’s the most popular. To get started with DDP,
```

File: docs/how-to/rocm-for-ai/train-a-model.rst [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+*  `Building a decoder transformer model on AMD GPUs — ROCm Blogs
+   <https://rocm.blogs.amd.com/artificial-intelligence/decoder-transformer/README.html#distributed-training-on-multiple-gpus>`_
+
+.. _rocm-for-ai-pytorch-fsdp:
+
+PyTorch FSDP
+------------
+
+As we saw in :ref:`PyTorch distributed <rocm-for-ai-pytorch-distributed>`, in DDP, model weights and optimizer states
```
Comment (lpaoletti) [inline docs/how-to/rocm-for-ai/train-a-model.rst:64]:
```suggestion
As noted in :ref:`PyTorch distributed <rocm-for-ai-pytorch-distributed>`, in DDP model weights and optimizer states
```

File: docs/how-to/rocm-for-ai/train-a-model.rst [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+See `Pre-training a large language model with Megatron-DeepSpeed on multiple AMD GPUs — ROCm Blogs
+<https://rocm.blogs.amd.com/artificial-intelligence/megatron-deepspeed-pretrain/README.html>`_ for a detailed example of
+training with DeepSpeed on an AMD accelerator or GPU.
+
+.. _rocm-for-ai-automatic-mixed-precision:
+
+Automatic mixed precision (AMP)
+-------------------------------
+
+As models increase in size, the time and memory needed to train them–that is, their cost–also increases. Any measure we
```
Comment (lpaoletti) [inline docs/how-to/rocm-for-ai/train-a-model.rst:97]:
```suggestion
As models increase in size, the time and memory needed to train them; that is, their cost also increases. Any measure we
```

File: docs/how-to/rocm-for-ai/train-a-model.rst [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+can take to reduce training time and memory usage through Automatic Mixed Precision (AMP) is highly beneficial for most
+use cases.
+
+See `Automatic mixed precision in PyTorch using AMD GPUs — ROCm Blogs
+<https://rocm.blogs.amd.com/artificial-intelligence/automatic-mixed-precision/README.html#automatic-mixed-precision-in-pytorch-using-amd-gpus>`_
+for more information on running AMP on an AMD accelerator.
```
Comment (lpaoletti) [inline docs/how-to/rocm-for-ai/train-a-model.rst:103]:
```suggestion
for more information about running AMP on an AMD accelerator.
```

File: docs/how-to/rocm-for-ai/hugging-face-models.rst (hunk 5 above)
Comment (peterjunpark) [reply docs/how-to/rocm-for-ai/hugging-face-models.rst:88]: This is correct. Ref: https://huggingface.co/docs/optimum/en/amd/installation

File: docs/how-to/rocm-for-ai/deploy-your-model.rst [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+Serving using vLLM
+==================
+
+vLLM is a fast and easy-to-use library for LLM inference and serving. vLLM officially supports ROCm versions 5.7 and
+6.0. AMD is actively working with the vLLM team to improve performance and support later ROCm versions.
+
+See the `GitHub repository <https://github.com/vllm-project/vllm>`_ and `official vLLM documentation
+<https://docs.vllm.ai/>`_ for more information.
+
+For guidance on using vLLM with ROCM, refer to `Installation with ROCm
```
Comment (yhuiYH) [inline docs/how-to/rocm-for-ai/deploy-your-model.rst:25]:
```suggestion
For guidance on using vLLM with ROCm, refer to `Installation with ROCm
```
Small typo

**Comment by peterjunpark** [comment]
@andyluo7 
Reordered the topics to:
1. Installation
2. Training a model
3. Running models from Hugging Face
4. Deploying your model

**Review by samjwu — APPROVED** [review]

**Review by lpaoletti — APPROVED** [review]

**Pull request merged**